resources/var/tmp/resources/init.groovy.d/scripts/init090NORMmavenautoinstall.groovy

```diff
@@ -46,32 +61,56 @@ def addMavenToInstallations(def installation) {
     Jenkins.instance.save()
 }
 
-def removeNonTargetM3Installations(def mavenName, def targetVersion) {
+// Checks if property has only one hudson.tasks.Maven$MavenInstaller installer and no other installers.
+// Other possible installers are hudson.tools.ZipExtractionInstaller,
+// hudson.tools.CommandInstaller or hudson.tools.BatchCommandInstaller.
+boolean propertyHasOneMavenInstallerOnly(def property) {
+    int numberOfMavenInstallers = 0
+    boolean noOtherInstallerDetected = true
+    property.installers.each { installer ->
+        if (installer instanceof hudson.tasks.Maven$MavenInstaller){
+            numberOfMavenInstallers++
+        } else {
+            noOtherInstallerDetected = false
+        }
+    }
+    return noOtherInstallerDetected && numberOfMavenInstallers == 1
+}
+
+// Gets Maven version as String if only one Maven installer is defined in property
+String getSingleMavenInstallerVersion(def property){
+    return property.installers[0].id
+}
+
+List getNonTargetM3Installations(def mavenName, def targetVersion) {
```

Comment:
We gathered the relevant installation before the call of this function. Reusing it will reduce the complexity of this function